File: code/70123.pas
program Bai123;
var n, i: integer;
    a :Array[1 .. 200] of integer;
begin
readln(n);
for i := 1 to n do read(a[i]);
for i := 2 to n do 
  if a[i]*a[i-1] >= 0 then  
    begin
    writeln('NO');
    exit
    end;
writeln('YES');
end.

File: code/70123.cpp
#include <iostream>
using namespace std;

int main() 
{
    int a[10000];
    int n;
    cin>>n;
    for (int i=0; i<=n-1; i++)
        cin>>a[i];
    for (int i=1; i<=n-1; i++){
        if (a[i]*a[i-1]>=0){
            cout<<"NO";
            return 0;
        }
    }
    cout<<"YES";
    return 0;
}